### DISCUSSION TITLE
ROS2 Galactic support

### DISCUSSION THREAD

**Discussion by alfred-wrigfors** [ouster-lidar/ouster-ros#135] [Q&A] [answered]
Hi!

We are part of the Formula Student Driverless competition, and are using some pre-built tools and simulators that are currently using Galactic.

Knowing that ROS2 versions sometimes have compatible packages, I tried to build this ROS2 driver/package for Galactic. It is currently not building. It is the ouster-ros node that fails on tf2_eigen/tf2_eigen.hpp.

Is there a known workaround or an older version that support Galactic?

Any help appreciated!

**Comment by Samahu** [comment, marked as answer]
Which branch have you used? We have two branches for ros2:

- ros rolling / humble:
https://github.com/ouster-lidar/ouster-ros/tree/ros2
- ros foxy:
https://github.com/ouster-lidar/ouster-ros/tree/ros2-foxy

Since galactic is older then you should check if it could work with the 'ros2-foxy’ branch but it isn’t clear from your question which branch have you tried.

**Reply by Samahu** [reply]
@alfred-wrigfors I have created this [pull request](https://github.com/ouster-lidar/ouster-ros/pull/136) which [verified](https://github.com/ouster-lidar/ouster-ros/actions/runs/4948764107/jobs/8849954024) that the project builds successfully against ros-galactic if you used the ros2-foxy branch. The fact it builds doesn't necessarily means it runs successfully but at least it is a good start and this ros2-foxy is the branch that you should be using when working with older distros. Note that we don't officially support ros galactic distro since it is past its time.

**Reply by alfred-wrigfors** [reply]
Hi!

Thank you! I cloned it and was also able to build it!

However, when running ti and visualizing it in rViz2, it seems like about 90 deg (last 90 deg) of the scan is lost. And, while trying to subscribe to /ouster/points, i get a warning the the QoS-settings are not compatible (and will not send any data to that subscriber).

Have done some research regarding how the QoS warning could be resolved, but with little to no success. Any pointers? The 90 deg-problem is at the moment not an issue that need to be solved for our applicatiion, just a nice-to-have.

Once again, thank you for you quick reply!

**Reply by alfred-wrigfors** [reply]
The warning:
`New publisher discovered on topic '/ouster/points', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY`

**Reply by Samahu** [reply]
For the 90 deg los, first make sure that you reset your azimuth window configuration on your sensor..(maybe you have accidentally changed it from the default when setting up the sensor).

For the RVIZ warning message, you will need to adjust the QoS settings of the topic in yor RVIZ such that it would match the QoS settings of the `/ouster/points`. Use `ros2 topic info /ouster/points` to check the current QoS settings (u may need to add -v for the command for extended results), then when launch RVIZ adjust the topic settings accordingly and the warning should go away.
